test/unit/connection_string.test.ts

```diff
@@ -832,4 +834,47 @@ describe('Connection String', function () {
         .that.matches(/useUnifiedTopology has no effect/);
     });
   });
+
+  describe('when mongodbLogPath is in options', function () {
+    const loggerFeatureFlag = Symbol.for('@@mdb.enableMongoLogger');
+
+    let stderrStub;
+    let stdoutStub;
+
+    before(() => {
+      stdoutStub = sinon.stub(process.stdout);
+      stderrStub = sinon.stub(process.stderr);
+    });
+
+    after(() => {
+      sinon.restore();
+    });
+
+    it('when option is `stderr`, it is accessible through mongoLogger.logDestination', function () {
+      const client = new MongoClient('mongodb://a/?mongodbLogPath=stderr', {
+        [loggerFeatureFlag]: true
+      });
+      const log: Log = { t: new Date(), c: 'ConnectionStringStdErr', s: 'error' };
+      client.options.mongoLoggerOptions.logDestination.write(log);
+      expect(stderrStub.write).calledWith(inspect(log, { breakLength: Infinity, compact: true }));
+    });
+
+    it('when option is `stdout`, it is accessible through mongoLogger.logDestination', function () {
+      const client = new MongoClient('mongodb://a/?mongodbLogPath=stdout', {
+        [loggerFeatureFlag]: true
+      });
+      const log: Log = { t: new Date(), c: 'ConnectionStringStdOut', s: 'error' };
+      client.options.mongoLoggerOptions.logDestination.write(log);
+      expect(stdoutStub.write).calledWith(inspect(log, { breakLength: Infinity, compact: true }));
+    });
+
+    it('when option is invalid, it defaults to stderr', function () {
+      const client = new MongoClient('mongodb://a/?mongodbLogPath=stdnothing', {
+        [loggerFeatureFlag]: true
+      });
+      const log: Log = { t: new Date(), c: 'ConnectionStringInvalidOption', s: 'error' };
+      client.options.mongoLoggerOptions.logDestination.write(log);
+      expect(stderrStub.write).calledWith(inspect(log, { breakLength: Infinity, compact: true }));
+    });
```

Comment:
If we have cases where the language uses "when", prefer using surrounding context blocks for the tests to better organise the output.

```suggestion
  context('when the option is stderr', function () {
      it('it is accessible through mongoLogger.logDestination', function () {
        const client = new MongoClient('mongodb://a/?mongodbLogPath=stderr', {
          [loggerFeatureFlag]: true
        });
        const log: Log = { t: new Date(), c: 'ConnectionStringStdErr', s: 'error' };
        client.options.mongoLoggerOptions.logDestination.write(log);
        expect(stderrStub.write).calledWith(inspect(log, { breakLength: Infinity, compact: true }));
      });
    });

    context('when the option is stdout', function () {  
      it('it is accessible through mongoLogger.logDestination', function () {
        const client = new MongoClient('mongodb://a/?mongodbLogPath=stdout', {
          [loggerFeatureFlag]: true
        });
        const log: Log = { t: new Date(), c: 'ConnectionStringStdOut', s: 'error' };
        client.options.mongoLoggerOptions.logDestination.write(log);
        expect(stdoutStub.write).calledWith(inspect(log, { breakLength: Infinity, compact: true }));
      });
    });
 
   context('when the option is invalid', function () {
      it('it defaults to stderr', function () {
        const client = new MongoClient('mongodb://a/?mongodbLogPath=stdnothing', {
          [loggerFeatureFlag]: true
        });
        const log: Log = { t: new Date(), c: 'ConnectionStringInvalidOption', s: 'error' };
        client.options.mongoLoggerOptions.logDestination.write(log);
        expect(stderrStub.write).calledWith(inspect(log, { breakLength: Infinity, compact: true }));
      });
    });
```

Reply:
Can we also add tests for providing a `MongoLogWritable` as well?

Reply:
Yes! I can, and I'll change the naming of the test blocks to fit conventions too.